python/tk_multi_reviewsubmission/renderer.py

```diff
@@ -1,187 +1,210 @@
 # Copyright (c) 2013 Shotgun Software Inc.
-# 
+#
 # CONFIDENTIAL AND PROPRIETARY
-# 
-# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
 # Source Code License included in this distribution package. See LICENSE.
-# By accessing, using, copying or modifying this work you indicate your 
-# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 import sgtk
 import os
 import sys
-import nuke
+
+import subprocess
+
+try:
+    import nuke
+except:
+    nuke = None
+
 
 class Renderer(object):
-    
     def __init__(self):
         """
         Construction
         """
-        self.__app = sgtk.platform.current_bundle() 
-        
-        self._burnin_nk = os.path.join(self.__app.disk_location, "resources", "burnin.nk")
+        self.__app = sgtk.platform.current_bundle()
         self._font = os.path.join(self.__app.disk_location, "resources", "liberationsans_regular.ttf")
-        
-        # If the slate_logo supplied was an empty string, the result of getting 
-        # the setting will be the config folder which is invalid so catch that
-        # and make our logo path an empty string which Nuke won't have issues with.
+
+        burnin_template = self.__app.get_template("burnin_path")
+        self._burnin_nk = burnin_template.apply_fields({})
+        # If a show specific burnin file has not been defined, take it from the default location
+        if not os.path.isfile(self._burnin_nk):
+            self._burnin_nk = os.path.join(self.__app.disk_location, "resources", "burnin.nk")
+
         self._logo = None
-        if os.path.isfile( self.__app.get_setting("slate_logo", "") ):
-            self._logo = self.__app.get_setting("slate_logo", "")
+        logo_template = self.__app.get_template("slate_logo")
+        if os.path.isfile(logo_template.apply_fields({})):
```

Comment:
Same comment as above about passing in context fields. Also, apply_fields is expensive enough we should only do it once...